File: modules/calendar-gaps/src/CalendarGapsModule.ts
import { requireOptionalNativeModule } from 'expo-modules-core';

import type { CalendarAuthorization, CalendarGap } from './types';

type CalendarGapsModuleType = {
  isAvailable(): boolean;
  requestAuthorization(): Promise<CalendarAuthorization>;
  findNextGap(
    minMinutes: number,
    withinSeconds: number,
  ): Promise<CalendarGap | null>;
};

// Optional: null when not linked (Expo Go / web / Jest).
export default requireOptionalNativeModule<CalendarGapsModuleType>(
  'HoneCalendarGapsModule',
);

File: modules/calendar-gaps/android/src/main/java/expo/modules/calendargaps/CalendarGapsModule.kt
package expo.modules.calendargaps

import expo.modules.kotlin.Promise
import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

/**
 * Android stub. Calendar gap-detection on Android requires READ_CALENDAR
 * permission and a CalendarContract query — out of scope for v1.1 per the
 * plan §"Out of scope for v1.1".
 */
class CalendarGapsModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("HoneCalendarGapsModule")

    Function("isAvailable") { -> false }

    AsyncFunction("requestAuthorization") { promise: Promise ->
      promise.resolve("denied")
    }

    AsyncFunction("findNextGap") { _minMinutes: Double, _withinSeconds: Double, promise: Promise ->
      promise.resolve(null)
    }
  }
}
